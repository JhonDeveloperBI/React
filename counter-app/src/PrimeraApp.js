import PropTypes from "prop-types";
//Functional Component


const PrimeraApp =( {saludo} ) =>{ //desestructuracion

    // if( !saludo){
    //     throw new Error(' EL saludo es necesario')
    // }

    return ( 
             <>
             <h1 > { saludo }</h1>
                {/* <pre> { JSON.stringify(saludo, null, 3) }</pre> */}
                <p> Mi primera aplicacion</p>
            </>
    );
}

PrimeraApp.propTypes = {
  saludo:PropTypes.string.isRequired   
}

export default PrimeraApp;