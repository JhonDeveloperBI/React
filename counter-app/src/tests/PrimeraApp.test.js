import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe('test in <PrimeraApp />', () => { 
   /*  test('should show message "Hola', () => { 

        const saludo = 'Hola';

        const { getByText } = render(<PrimeraApp saludo={ saludo } />)

        expect( getByText(saludo)).toBeInTheDocument();
     }) */
     
     test('should be show <PrimeraApp /> success',() => {

        const saludo = 'Hola' 
        const wrapper = shallow(<PrimeraApp saludo={ saludo }/>) // simulate click, documents
         
        expect( wrapper).toMatchSnapshot(); // snapshot
     });


     test('should show the subtitle send for props',() => {

      const saludo = 'Hola';
      const subtitulo = 'mensaje';

      const wrapper = shallow(
                  <PrimeraApp 
                  saludo={ saludo }
                  subtitulo = { subtitulo }
                  />
      ); // simulate click, documents
       
      const textParrafo = wrapper.find('p').text(); //document.querySelector

      
      expect(textParrafo).toBe(subtitulo); // snapshot
   });

 })