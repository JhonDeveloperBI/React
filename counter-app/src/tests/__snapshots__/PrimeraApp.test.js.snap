// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <PrimeraApp /> should be show <PrimeraApp /> success 1`] = `
<Fragment>
  <h1>
     
    Hola
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
